tsl/test/sql/cagg_api.sql

```diff
@@ -0,0 +1,158 @@
+
+INSERT INTO hyper_ts
+  SELECT time, ceil(random() * 100)::int
+    FROM generate_series('2025-01-01'::timestamp,
+                         '2025-01-06', '1m') time;
+
+INSERT INTO hyper_tstz
+  SELECT time, ceil(random() * 100)::int
+    FROM generate_series('2025-01-01'::timestamptz,
+                         '2025-01-06', '1m') time;
+
+CREATE MATERIALIZED VIEW ts_temperature_1h
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('1 hour', time), avg(value)
+  FROM hyper_ts
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW ts_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_ts
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW tstz_temperature_1h
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('1 hour', time), avg(value)
+  FROM hyper_tstz
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW tstz_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_tstz
+GROUP BY 1;
+
+CREATE MATERIALIZED VIEW multi_temperature_15m
+  WITH  (timescaledb.continuous) AS
+SELECT time_bucket('15 minutes', time), avg(value)
+  FROM hyper_multi
+GROUP BY 1;
+
+SET search_path TO _timescaledb_functions, public;
+
+-- These are not part of the API, but we test them here just to make
+-- sure they work as expected.
+SELECT table_name, get_materialization_info(table_name)
+  FROM (
+      VALUES ('tstz_temperature_15m'), ('multi_temperature_15m')
+  ) t(table_name);
+
+\set ON_ERROR_STOP 0
+SELECT get_materialization_info('hyper_no_cagg');
+\set ON_ERROR_STOP 1
+
+-- This is not part of the API either, but added a test here to make
+-- sure that it works as expected.
+SELECT materialization_id,
+       to_timestamp(lowest_modified_value),
+       to_timestamp(greatest_modified_value)
+  FROM get_raw_materialization_ranges('timestamptz');
+
+-- Here are tests of the API
+SELECT *
+  INTO before
+  FROM _timescaledb_catalog.continuous_aggs_materialization_invalidation_log;
+
+CALL _timescaledb_functions.add_materialization_invalidations(
+     'tstz_temperature_15m'::regclass,
+     '["2025-04-25 11:10:00+02","2025-04-26 11:14:00+02"]'::tstzrange
+);
+
+CALL _timescaledb_functions.add_materialization_invalidations(
+     'ts_temperature_15m'::regclass,
+     '["2025-04-25 11:10:00+02","2025-04-26 11:14:00+02"]'::tsrange
+);
+
+-- Custom refresh function that iterate over the ranges inside the
+-- restriction and refresh them. This is to check that the refresh
+-- function does the right thing with the ranges returned by the API
+-- function.
+CREATE PROCEDURE custom_refresh(cagg REGCLASS, restriction ANYRANGE) AS
+$body$
+DECLARE
+  inval restriction%TYPE;
+BEGIN
+   FOR inval IN
+   SELECT UNNEST(invalidations)
+     FROM _timescaledb_functions.get_materialization_invalidations(cagg, restriction)
+   LOOP
+       RAISE NOTICE 'Updating range %', inval;
+       CALL refresh_continuous_aggregate(cagg, lower(inval), upper(inval));
```

Comment:
```suggestion
       CALL refresh_continuous_aggregate(cagg, lower(inval), upper(inval));
       COMMIT;
```
I think after this procedure call you should add a commit, no?

Reply:
I think the current tests are not covering more than one invalidation to be refreshed or am I missed something?

Reply:
> I think after this procedure call you should add a commit, no?

Since it is just for the test, it makes no real difference (`CALL` will not allow executing transaction statements inside a transaction and refresh does an commit internally, so AFAICT, it would fail in this case), but did it anyway.

> I think the current tests are not covering more than one invalidation to be refreshed or am I missed something?

They are just using a single invalidation, correct.